src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -223,10 +223,12 @@ private DRSContentsObject makeDrsContentsObject(FSItem fsObject, String snapshot
 
     private String makeHttpsFromGs(String gspath) {
         try {
-            URI gsuri = URI.create(gspath);
-            String gsBucket = gsuri.getAuthority();
-            String gsPath = StringUtils.removeStart(gsuri.getPath(), "/");
-            String encodedPath = URLEncoder.encode(gsPath, StandardCharsets.UTF_8.toString());
+            String gsBucket = GcsPdao.extractBucketFromPath(gspath);
+            String gsPath = GcsPdao.extractFilePathInBucket(gspath, gsBucket);
+            String encodedPath = URLEncoder.encode(gsPath, StandardCharsets.UTF_8.toString())
```

Comment:
I just did some reading and i think i understand the root of the problem. URLEncoder.encode complies with the  `application/x-www-form-urlencoded` MIME format which is not the same as encoding for a URI.

We might get better results using the [URI](https://docs.oracle.com/javase/7/docs/api/java/net/URI.html) class instead of the encoder class.

Reply:
Ah interesting.  Let me try that

Reply:
Ah, no luck.  This ended up encoding the path separators which caused other issues :(

Going to leave as is